libraries/botbuilder-ai-orchestrator/src/orchestratorAdaptiveRecognizer.ts

```diff
@@ -253,6 +291,67 @@ export class OrchestratorAdaptiveRecognizer
         return recognizerResult;
     }
 
+    private getTopTwoIntents(
+        result: RecognizerResult
+    ): { name: string; score: number }[] {
+        if (!result || !result.intents) {
+            throw new Error('result is empty');
+        }
+        const intents = Object.entries(result.intents)
+                            .map((intent) => {return {name: intent[0], score: +intent[1].score} })
+                            .sort((a,b) => (b.score - a.score))
+        intents.length = 2;
+
+        return intents;
+    }
+
+    /**
+     * Uses the RecognizerResult to create a list of properties to be included when tracking the result in telemetry.
+     * 
+     * @param {RecognizerResult} recognizerResult Recognizer Result.
+     * @param {Record<string, string>} telemetryProperties A list of properties to append or override the properties created using the RecognizerResult.
+     * @param {DialogContext} dialogContext Dialog Context.
+     * @returns {Record<string, string>} A collection of properties that can be included when calling the TrackEvent method on the TelemetryClient.
+     */
+    protected fillRecognizerResultTelemetryProperties(
+        recognizerResult: RecognizerResult,
+        telemetryProperties: Record<string, string>,
+        dialogContext?: DialogContext
+    ): Record<string, string> {
+        const topTwo = this.getTopTwoIntents(recognizerResult);
+        const intents = Object.entries(recognizerResult.intents);
+        const properties: Record<string, string> = {
+            TopIntent: intents.length > 0 ? topTwo[0].name : undefined,
+            TopIntentScore: intents.length > 0 ? topTwo[0].score.toString() : undefined,
+            NextIntent: intents.length > 1 ? topTwo[1].name : undefined,
+            NextIntentScore: intents.length > 1 ? topTwo[1].score.toString() : undefined,
+            Intents: intents.length > 0 ? JSON.stringify(recognizerResult.intents) : undefined,
+            Entities: recognizerResult.entities ? JSON.stringify(recognizerResult.entities) : undefined,
+            AdditionalProperties: JSON.stringify(
+                omit(recognizerResult, ['text', 'alteredText', 'intents', 'entities'])
```

Comment:
`omit` can generally be replaced by doing:

```typescript
const { omit, keys, here, ...valueWithoutKeys } = value;
```

Reply:
Thanks, in latest branch..